app/src/renderer/vuex/modules/send.js

```diff
@@ -62,94 +60,69 @@ export default ({ node }) => {
-        return node[type](to, args)
-      } else {
-        return node[type](args)
-      }
-    },
-    async sendTx({ state, dispatch, commit, rootState }, args) {
-      if (!rootState.connection.connected) {
-        throw Error(
-          `Currently not connected to a secure node. Please try again when Voyager has secured a connection.`
-        )
-      }
-
-      await dispatch(`queryWalletBalances`) // the nonce was getting out of sync, this is to force a sync
-
-      const gasPrices = [
-        {
-          amount: `0.025`, // recommended on Cosmos Docs
-          denom: rootState.stakingParameters.parameters.bond_denom || `uatom`
-        }
-      ]
-      // TODO: run gas estimations to optimize the value of the adjustment
-      const gasAdjustment = 1.5
+    cleanRequestArguments({ state, rootState }, args) {
 
       const requestMetaData = {
         sequence: state.nonce,
         from: rootState.wallet.address,
         account_number: rootState.wallet.accountNumber,
         chain_id: rootState.connection.lastHeader.chain_id,
-        gas: ``, // set after simulation
-        gas_prices: gasPrices,
-        gas_adjustment: String(gasAdjustment),
-        simulate: true,
+        gas: args.gas,
+        gas_prices: args.gas_prices,
+        simulate: args.simulate,
         memo: `Sent via Cosmos Wallet 🚀`
       }
-      args.base_req = requestMetaData
 
       // extract type
-      const type = args.type || `send`
-      delete args.type
+      const type = args.type
 
       // get signing method: localkeystore or ledger
-      const submitType = args.submitType || `local`
-      delete args.submitType
+      const submitType = args.submitType || `ledger`
 
       // TODO: refactor to get path request parameters at once
       // extract path parameters
       const to = args.to
       const pathParameter = args.pathParameter
-      delete args.to
-      delete args.pathParameter
-
-      // simulation to get the estimated gas
-      let request = actions.apiRequest(type, to, pathParameter, args)
-      const simulationRes = await request.catch(handleSDKError)
-      const adjustedGas = Number(simulationRes.gas_estimate) * gasAdjustment
-      const estimatedFeesLow =
-        Number(simulationRes.gas_estimate) * Number(gasPrices[0].amount) * 1e-6
-      const estimatedFeesHigh =
-        Number(adjustedGas) * Number(gasPrices[0].amount) * 1e-6
-
-      // TODO: remove once fees are displayed on action modals
-      console.log(
-        `- Estimated Gas: ${simulationRes.gas_estimate}\n` +
-        `- Adjusted Gas: ${adjustedGas}\n` +
-        `- Estimated Fees (low): ${estimatedFeesLow} ${gasPrices[0].denom}s\n` +
-        `- Estimated Fees (high): ${estimatedFeesHigh} ${gasPrices[0].denom}s`
-      )
-      args.base_req.simulate = false
-      args.base_req.gas = simulationRes.gas_estimate // adjusted on the SDK
+      const properties = [`submitType`, `type`, `to`, `pathParameter`, `gas`, `gas_prices`, `simulate`]
```

Comment:
code smell

Reply:
?

Reply:
manipulating this `args` object is not how we should do things. as it is dangerous as we might forget properties.